### PR TITLE
Feature/aww 1229 1230 1401 1399 new tx atomic swap transfer money, 

### PR DESCRIPTION
**Pull request by leonidstr**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/transaction/DexTransferMoneyTransaction.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Override
+    public AbstractAttachment parseAttachment(JSONObject attachmentData) throws AplException.NotValidException {
+        return new DexControlOfFrozenMoneyAttachment(attachmentData);
+    }
+
+    @Override
+    public void validateAttachment(Transaction transaction) throws AplException.ValidationException {
+        if (transaction.getAmountATM() <= 0 || transaction.getAmountATM() >= blockchainConfig.getCurrentConfig().getMaxBalanceATM()) {
+            throw new AplException.NotValidException("Invalid ordinary payment");
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/transaction/DexTransferMoneyTransaction.java]: Replace "Invalid ordinary payment" by something else

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/ParameterParser.java [hunk 2]
```diff
@@ -1019,6 +1026,71 @@ public static TwoFactorAuthParameters parse2FARequest(HttpServletRequest req, St
...
+        }
+
+        if(phasingFinishHeight != -1 && phasingTimeLockDuration != -1){
+            throw new ParameterException(
+                    JSONResponses.incorrect("Only one parameter should be filled 'phasingFinishHeight or phasingFinishTime'"));
+        }
+
+        int phasingFinishTime = -1;
+        if(phasingTimeLockDuration == -1) {
+            phasingFinishHeight = ParameterParser.getInt(req, "phasingFinishHeight",
```
[the remark this answers is not in the document]
Comment (leonidstr) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/ParameterParser.java]: I use inner validation here.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java [hunk 3]
```diff
@@ -214,5 +227,37 @@ public boolean isThereAnotherCancelUnconfirmedTx(Long orderId, Long cancelTrId){
...
+        Transaction transaction = null;
+        try {
+            transaction = dexOfferTransactionCreator.createTransaction(createTransactionRequest);
+
+        } catch (AplException.ValidationException e) {
+            LOG.error(e.getMessage(), e);
+            //TODO
+        } catch (ParameterException e) {
+            LOG.error(e.getMessage(), e);
+            //TODO
```
[the remark this answers is not in the document]
Comment (leonidstr) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java]: I will take a look on it then.
Right now it's difficult to say.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java [hunk 4]
```diff
@@ -214,5 +227,37 @@ public boolean isThereAnotherCancelUnconfirmedTx(Long orderId, Long cancelTrId){
...
+        createTransactionRequest.setPhasing(phasing);
+
+        createTransactionRequest.setAttachment(new DexControlOfFrozenMoneyAttachment(offer.getTransactionId() != null ? offer.getTransactionId() : 0 , false));
+        Transaction transaction = null;
+        try {
+            transaction = dexOfferTransactionCreator.createTransaction(createTransactionRequest);
+
+        } catch (AplException.ValidationException e) {
+            LOG.error(e.getMessage(), e);
+            //TODO
```
[the remark this answers is not in the document]
Comment (leonidstr) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java]: I will take a look on it then.
Right now it's difficult to say.

**Pull request merged**